appeals/web/src/server/appeals/appeal-details/lpa-questionaire/lpa-questionnaire.controller.js

```diff
@@ -1,49 +1,69 @@
+import * as lpaQuestionnaireService from './lpa-questionnaire.service.js';
 import {
-	getLpaQuestionnaireFromId,
-	getLPAQuestionnaireIncompleteReasons,
-	setReviewOutcomeForLpaQuestionnaire
-} from './lpa-questionnaire.service.js';
-import {
-	mapLpaQuestionnaire,
+	backLink,
+	lpaQuestionnairePage,
 	mapReviewOutcomeToSummaryListBuilderParameters,
-	mapWebReviewOutcomeToApiReviewOutcome
+	mapWebReviewOutcomeToApiReviewOutcome,
+	pageTitle
 } from './lpa-questionnaire.mapper.js';
 import { generateSummaryList } from '../../../lib/nunjucks-template-builders/summary-list-builder.js';
 import logger from '../../../lib/logger.js';
 import * as appealDetailsService from '../appeal-details.service.js';
 import { objectContainsAllKeys } from '../../../lib/object-utilities.js';
 import { renderDecisionIncompleteConfirmationPage } from './outcome-incomplete/outcome-incomplete.controller.js';
+import { appealShortReference } from '#lib/nunjucks-filters/appeals.js';
 
 /**
- *
- * @param {string} appealId
- * @param {string} lpaQId
+ * @param {import('@pins/express/types/express.js').Request} request
  * @param {import("@pins/express/types/express.js").ValidationErrors | string | null} errors
  * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
- * @param {import('got').Got} apiClient
  */
-const renderLpaQuestionnaire = async (apiClient, appealId, lpaQId, response, errors = null) => {
-	const lpaQuestionnaireResponse = await getLpaQuestionnaireFromId(apiClient, appealId, lpaQId);
+const renderLpaQuestionnaire = async (request, response, errors = null) => {
+	let lpaQuestionnairePromise = lpaQuestionnaireService.getLpaQuestionnaireFromId(
+		request.apiClient,
+		request.params.appealId,
+		request.params.lpaQId
+	);
+	let appealDetailsPromise = appealDetailsService.getAppealDetailsFromId(
+		request.apiClient,
+		request.params.appealId
+	);
 
-	const mappedLpaQuestionnaireSections = mapLpaQuestionnaire(lpaQuestionnaireResponse);
-	const formattedSections = [];
-	for (const section of mappedLpaQuestionnaireSections) {
-		formattedSections.push(generateSummaryList(section));
-	}
+	const [lpaQuestionnaire, appealDetails] = await Promise.all([
+		lpaQuestionnairePromise,
+		appealDetailsPromise
+	]);
+	const session = request.session;
 
-	return response.render('appeals/appeal/lpa-questionnaire.njk', {
-		summaryList: { formattedSections },
-		errors
-	});
+	if (lpaQuestionnaire && appealDetails) {
+		const currentUrl = request.protocol + '://' + request.get('host') + request.originalUrl;
+		const pageComponents = lpaQuestionnairePage(
+			{ lpaQ: lpaQuestionnaire },
```

Comment:
You might be able to avoid needing to wrap the lpaq and appealDetails in these objects if you refactor `lpaQuestionnairePage` to accept them as-is